### ISSUE TITLE
Authentication in RTMP 

### ISSUE THREAD

**Issue by dinesh547** [jgh-/VideoCore#201]
Seems like there is no authentication support has been implemented in current code.

**Comment by jgh-** [comment]
There is not, but if somebody wants to implement it that would be a great PR.

**Comment by dinesh547** [comment]
James, I am trying to implement the same but there are no much documentation available for RTMP. Do you know any document which details the authentication in RTMP and RTMP as whole as well..

**Comment by shubhank008** [comment]
Pretty sure you can just use rtmp://user:password@host  
It works for FFMpeg atleast

**Comment by dinesh547** [comment]
@shubhank008 as per i know it works only for ffmpeg (or applications using ffmpeg) not for videocore.

**Comment by jgh-** [comment]
Not really, when I was developing the RTMP stuff originally I was mainly going by the "spec" that Adobe released a few years back.

**Comment by dinesh547** [comment]
Hi @jgh-  i am trying to implement the authentication in your code. But i am facing some issues while sending the data.. 
What i am doing is checking if I am getting an authentication failure message in the    
 void RTMPSession::handleInvoke(uint8_t* p) inside RTMPSession.cpp class.
The Authentication failure message comes as in the handleInvoke function
 so i checked if (command == "_error") and than i am checking inside the if statement 
        std::string code = parseStatusCode(p + 3 + command.length());
            if(code == "NetConnection.Connect.Rejected") {
//Authentication failure
sendCallAuthenticationRequest()
} 

Now when is send the second connect request from inside this function i never receives any response. and i am not sure why it is happening.. (Code is not in pure objective c and i am not very good in c++ OOPS). Can you help me in this.. Right now i am doing authentication for adobe (authmod=adobe)..

**Comment by keithcml** [comment]
@dinesh547 can you give more details to me, I can help to make the authentication feature. I am now have no idea about it.

**Comment by triwav** [comment]
Not sure if it's bad form but just want throw out there that I moved away from this project due to the lack of authentication. However I just had a ticket closed on another library that now correctly handles authentication Akamai which we use https://github.com/shogo4405/lf.swift/issues/55. If you need authentication might look into it.

**Comment by dinesh547** [comment]
ANybody who needs to implement authentication can refer to ffmpeg code. The authentication code is pretty much clear there.
